src/ort_genai_c.h

```diff
@@ -97,6 +97,12 @@ OGA_EXPORT const char* OGA_API_CALL OgaResultGetError(const OgaResult* result);
 OGA_EXPORT OgaResult* OGA_API_CALL OgaSetLogBool(const char* name, bool value);
 OGA_EXPORT OgaResult* OGA_API_CALL OgaSetLogString(const char* name, const char* value);
 
+/**
+ * \param[in] callback function pointer to the logging callback function (use nullptr to disable callback)
+ * \return OgaResult containing the error message when the callback could not be set, else nullptr.
+ */
+OGA_EXPORT OgaResult* OGA_API_CALL OgaSetLogCallback(void (*callback)(const char* string, size_t length));
```

Comment:
if we are exposing a way to set an onnxruntime-genai log output function, would it make sense to also allow users to do this for the underlying onnxruntime logging? something to consider, not necessarily for this PR.

Reply:
yes. for foundry local it would be great to use the same logging callback for everything as we can't get the native logs from genai or ort easily.